packages/core/admin/admin/src/pages/Home/HomePage.tsx

```diff
@@ -55,14 +57,88 @@ export const WidgetRoot = ({ title, icon = PuzzlePiece, children }: RootProps) =
 };
 
 /* -------------------------------------------------------------------------------------------------
- * HomePageCE
+ * UnstableHomePageCe
  * -----------------------------------------------------------------------------------------------*/
 
+const UnstableHomePageCe = () => {
+  const { formatMessage } = useIntl();
+  const user = useAuth('HomePageCE', (state) => state.user);
+  const displayName = user?.firstname ?? user?.username ?? user?.email;
+  const getAllWidgets = useStrapiApp('UnstableHomepageCe', (state) => state.widgets.getAll);
+
+  const [loadedWidgets, setLoadedWidgets] = React.useState<
+    (Omit<ReturnType<typeof getAllWidgets>[number], 'component'> & {
+      component: React.ComponentType;
+    })[]
+  >([]);
+
+  React.useEffect(() => {
+    const loadComponents = async () => {
+      const widgets = getAllWidgets();
+
+      const loaded = await Promise.all(
+        widgets.map(async (widget) => {
+          const LazyComponent = await widget.component();
+
+          return {
+            ...widget,
+            component: LazyComponent,
+          };
+        })
+      );
+
+      setLoadedWidgets(loaded);
+    };
+
+    loadComponents();
+  }, [getAllWidgets]);
+
+  return (
+    <Main>
+      <Page.Title>
+        {formatMessage({ id: 'HomePage.head.title', defaultMessage: 'Homepage' })}
+      </Page.Title>
+      <Layouts.Header
+        title={formatMessage(
+          { id: 'HomePage.header.title', defaultMessage: 'Hello {name}' },
+          { name: displayName }
+        )}
+        subtitle={formatMessage({
+          id: 'HomePage.header.subtitle',
+          defaultMessage: 'Welcome to your administration panel',
+        })}
+      />
+      <Layouts.Content>
+        <Flex direction="column" alignItems="stretch" gap={8} paddingBottom={10}>
+          <GuidedTour />
+          <Grid.Root gap={5}>
+            {loadedWidgets.map((widget) => {
```

Comment:
You're mapping on an array that's only populated after a Promise.all, so I don't see how this wouldn't block all rendering?

I assume it's not very noticeable because it's fast, but I think we should provide lazy loading still, because some widgets could have big bundle sizes, if they require dataviz libraries for example. Also I think our ability to render Widget.Loading while waiting is neat, we should make the most of it

Reply:
I had [the same thought](https://github.com/strapi/strapi/pull/23352#discussion_r2039836719), I tested it this morning thought with a slow network connection. We see the title and the loading state for the content. But probably its just more readable/understandable to handle the components separately. I'll update.